functests/vrf/vrf.go

```diff
@@ -0,0 +1,276 @@
+package vrf
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	apiclient "github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	netattdefv1 "github.com/openshift/sriov-network-operator/pkg/apis/k8s/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	sameNode                     = "Same Node"
+	diffNode                     = "Different Node"
+	ipStackIPv4                  = "ipv4"
+	ipStackIPv6                  = "ipv6"
+	hostnameLabel                = "kubernetes.io/hostname"
+	TestNamespace                = "vrf-testing"
+	podWaitingTime time.Duration = 2 * time.Minute
+	VRFBlueName                  = "blue"
+	VRFRedName                   = "red"
+)
+
+var (
+	podClientVRFBlueIPAddress string
+	podServerVRFBlueIPAddress string
+	nodeParameters            = []string{sameNode, diffNode}
+	ipStackParameters         = []string{ipStackIPv4, ipStackIPv6}
+)
+
+type vrfTestParameters struct {
+	Node    string
+	IPStack string
+}
+
+var _ = Describe("[vrf]", func() {
+	describe := func(desc string) func(node string, ipStack string) string {
+
+		return func(node string, ipStack string) string {
+			VRFParameters, err := newVRFTestParameters(node, ipStack)
+			if err != nil {
+				return fmt.Sprintf("error in parameters: %s node=%s, ipStack=%s", desc, node, ipStack)
+			}
+			params, err := json.Marshal(VRFParameters)
+			if err != nil {
+				return fmt.Sprintf("error in parameters: %s node=%s, ipStack=%s", desc, node, ipStack)
+			}
+
+			return fmt.Sprintf("%s %s", desc, string(params))
+		}
+	}
+	apiclient := apiclient.New("")
+
+	var nodesList []k8sv1.Node
+	var vrfBlue netattdefv1.NetworkAttachmentDefinition
+	var vrfRed netattdefv1.NetworkAttachmentDefinition
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+		nodesList, err = nodes.GetByRole(apiclient, "worker")
+		Expect(err).ToNot(HaveOccurred())
+		defIntName := getMainNodeInterface(apiclient, nodesList[0].Name)
+		Expect(err).ToNot(HaveOccurred())
+		vrfBlue = addVrfNad("test-vrf-blue", defIntName, VRFBlueName)
```

Comment:
If addVrfNad only defines it, let's rename it.
Or alternative, let's embed create in the function.

One nit: should be VRF as per go guidelines on acronyms